File: threads/src/main/java/im/threads/internal/model/DateRow.java
package im.threads.internal.model;

import androidx.core.util.ObjectsCompat;

public final class DateRow implements ChatItem, MediaAndFileItem {
    private final long date;

    public DateRow(long date) {
        this.date = date;
    }

    @Override
    public long getTimeStamp() {
        return date;
    }

    public long getDate() {
        return date;
    }

    @Override
    public String toString() {
        return "DateRow{" +
                "date=" + date +
                '}';
    }

    @Override
    public boolean isTheSameItem(ChatItem otherItem) {
        return otherItem instanceof DateRow;
    }

    @Override
    public Long getThreadId() {
        return null;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        DateRow dateRow = (DateRow) o;
        return date == dateRow.date;
    }

    @Override
    public int hashCode() {
        return ObjectsCompat.hash(date);
    }
}

File: threads/src/main/java/im/threads/business/models/DateRow.kt
package im.threads.business.models

import androidx.core.util.ObjectsCompat

class DateRow(override val timeStamp: Long) : ChatItem, MediaAndFileItem {
    override fun toString(): String {
        return "DateRow{" +
            "date=" + timeStamp +
            '}'
    }

    override fun isTheSameItem(otherItem: ChatItem?): Boolean {
        return otherItem is DateRow
    }

    override val threadId: Long?
        get() = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val dateRow = other as DateRow
        return timeStamp == dateRow.timeStamp
    }

    override fun hashCode(): Int {
        return ObjectsCompat.hash(timeStamp)
    }
}
